### PR TITLE
[llvm-c] Add getters for LLVMContextRef for various types

### PR DESCRIPTION
**Pull request by abgeana**
Small PR to add additional getters for LLVMContextRef in the C API.

### CODE REVIEW

File: llvm/include/llvm-c/Core.h [hunk 1]
```diff
@@ -3434,6 +3441,13 @@ void LLVMMoveBasicBlockBefore(LLVMBasicBlockRef BB, LLVMBasicBlockRef MovePos);
  */
 void LLVMMoveBasicBlockAfter(LLVMBasicBlockRef BB, LLVMBasicBlockRef MovePos);
 
+/**
+ * Obtain the context to which this basic block is associated.
+ *
+ * @see llvm::Module::getContext()
+ */
+LLVMContextRef LLVMGetBasicBlockContext(LLVMBasicBlockRef BB);
```
Comment (nikic) [inline llvm/include/llvm-c/Core.h:3449]: This shouldn't be needed. A BasicBlock is also a Value.

**Review by nikic — COMMENTED** [review]
What do you need this for?

File: llvm/include/llvm-c/Core.h (hunk 1 above)
Comment (abgeana) [reply llvm/include/llvm-c/Core.h:3449]: I am planning to use this via [llvm-sys.rs](https://gitlab.com/taricorp/llvm-sys.rs), and that would make the API more consistent both in C and in the rust bindings.

File: llvm/include/llvm-c/Core.h [hunk 2]
```diff
@@ -3434,6 +3441,13 @@ void LLVMMoveBasicBlockBefore(LLVMBasicBlockRef BB, LLVMBasicBlockRef MovePos);
  */
 void LLVMMoveBasicBlockAfter(LLVMBasicBlockRef BB, LLVMBasicBlockRef MovePos);
 
+/**
+ * Obtain the context to which this basic block is associated.
+ *
+ * @see llvm::Module::getContext()
```
Comment (nikic) [inline llvm/include/llvm-c/Core.h:3447]:
```suggestion
 * @see llvm::BasicBlock::getContext()
```

File: llvm/include/llvm-c/Core.h (hunk 1 above)
Comment (nikic) [reply llvm/include/llvm-c/Core.h:3449]: Just to be clear, my point here is that you could write `LLVMValueGetContext(LLVMBasicBlockAsValue(BB))` to achieve the same -- do we really need a shorthand for it?

**Review by nikic — COMMENTED** [review]
Looks reasonable to me, but needs release note entries.

File: llvm/include/llvm-c/Core.h (hunk 1 above)
Comment (abgeana) [reply llvm/include/llvm-c/Core.h]: Was not aware initially of this, but yeah it makes sense to not include the additional function. I have removed it in the latest version.

File: llvm/include/llvm-c/Core.h (hunk 2 above)
Comment (abgeana) [reply llvm/include/llvm-c/Core.h]: Have removed the function and the associated comment.

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**